src/rplidar_node.cpp

```diff
@@ -157,36 +158,36 @@ void rplidar_node::publish_scan(
     scan_msg.angle_max = M_PI - angle_max;
   }
   scan_msg.angle_increment =
-    (scan_msg.angle_max - scan_msg.angle_min) / (double)(node_count);
+    (scan_msg.angle_max - scan_msg.angle_min) / (double)(node_count - 1);
```

Comment:
@chrisl8 wasn't this changed to be correct the other way? I'm confused now

Reply:
I was working with angle_compensated mode, in slam_toolbox, karto needs a consistent number of scan readings to be happy, the non-angle_compensated mode was giving variable readings each iteration.  Angle_compensated is giving 1 degree readings from [0,359] (turned into [-179,180]), so 360 readings.  Without the (node_count-1), it is reporting the angle increment (using degrees for readability) as 359/360 degrees, as opposed to the 1 degree expected.  I am not sure if that was a work-around for the threshold issue I fixed in slam_toolbox, as Karto is doing:
```
m_NumberOfRangeReadings = static_cast<kt_int32u>(math::Round((GetMaximumAngle() -
      GetMinimumAngle()) /
      GetAngularResolution()) + residual)
```
Where residual defaults to 1, unless a laser is reported to Karto as a 360 laser, but this is generally intended when a laser goes from [-PI,PI] and you don't want to count the beginning and end reading twice.  Because angle_compensated mode is reporting as [-179,180], it is not wrapping around, but the previous slam_toolbox check for 360 laser was doing ((max_angle - min_angle) - 2PI) < 0.1, which was incorrectly reporting [-179,180] as a 360 Lidar, so that led residual to be 0 in the equation above and would have led Karto to expect 359 readings.  Therefore, I can see making angle_increment report as 359/360 would have led the Karto math to report the expected 360 readings.  Now that slam_toolbox is fixed, this laser driver can use the correct angle_increment reporting 359/359 and the residual on Karto is 1 and it expects 360 readings.

Reply:
I have not tested all of this yet, but my understanding was that @justinIRBT created a superior fix in place of my hack.

He was also able to convince the slam_toolbox maintainer to alter their code, which I was unable to do. Probably because my explanation was poor and I didn't submit a PR.

I will have to back port this to the ROS1 driver before I can test it.

Reply:
FYI: I confirmed just now in testing that this update to slam_toolbox works with the ROS1 RPlidar driver provided by Slamtec.
That doesn't really have any affect on your merging of this PR for the ROS2 driver, but it does mean that updating the ROS1 driver isn't highly important.